### PR TITLE
[TRAFODION-2385] odbc support boolean,tinyint,largeint unsigned

### PR DESCRIPTION
**Pull request by Weixin-Xu**

### CODE REVIEW

File: core/conn/unixodbc/odbc/odbcclient/unixcli/cli/cdesc.cpp [hunk 1]
```diff
@@ -2381,7 +2381,8 @@ SQLRETURN CDesc::BuildValueList(CStmt *pHandle,  //AMR - changed this pointer ty
 
 				if(IPDDescRecPtr->m_ODBCDataType == SQL_VARCHAR ||
 				   IPDDescRecPtr->m_ODBCDataType == SQL_LONGVARCHAR ||
-				   IPDDescRecPtr->m_ODBCDataType == SQL_CHAR)
+				   IPDDescRecPtr->m_ODBCDataType == SQL_CHAR &&
+                   IPDDescRecPtr->m_SQLDataType != SQLTYPECODE_BOOLEAN)
```
Comment (DaveBirdsall) [inline core/conn/unixodbc/odbc/odbcclient/unixcli/cli/cdesc.cpp]: Is this the operator precedence you want? As written, this code means:
if ( VARCHAR or LONG VARCHAR or ( CHAR and not BOOLEAN) )
Perhaps you meant,
if ( ( VARCHAR or LONG VARCHAR or CHAR) and BOOLEAN) ) ?

File: win-odbc64/odbcclient/drvr35/cdesc.cpp [hunk 2]
```diff
@@ -2277,7 +2277,8 @@ SQLRETURN CDesc::BuildValueList(CStmt *pHandle,  //AMR - changed this pointer ty
 
 				if(IPDDescRecPtr->m_ODBCDataType == SQL_VARCHAR ||
 				   IPDDescRecPtr->m_ODBCDataType == SQL_LONGVARCHAR ||
-				   IPDDescRecPtr->m_ODBCDataType == SQL_CHAR)
+				   IPDDescRecPtr->m_ODBCDataType == SQL_CHAR &&
+				   IPDDescRecPtr->m_SQLDataType != SQLTYPECODE_BOOLEAN)
```
Comment (DaveBirdsall) [inline win-odbc64/odbcclient/drvr35/cdesc.cpp]: Same comment as above

File: core/conn/unixodbc/odbc/odbcclient/unixcli/cli/cdesc.cpp (hunk 1 above)
Comment (Weixin-Xu) [inline core/conn/unixodbc/odbc/odbcclient/unixcli/cli/cdesc.cpp]: when SQLDataType return SQLTYPECODE_BOOLEAN, ODBCDataType perform SQL_CHAR then code will make len-1.
It's unneccessary operation, so skip these code when meeting boolean type.
So VARCHAR or LONG VARCHAR or ( CHAR and not BOOLEAN)  is my meaning, and add () will be better

Comment (DaveBirdsall) [inline core/conn/unixodbc/odbc/odbcclient/unixcli/cli/cdesc.cpp]: Thanks for making this clear. Looks good.

File: win-odbc64/sql/cli/sqlcli.h [hunk 3]
```diff
@@ -425,6 +433,8 @@ enum SQLTYPE_CODE {
     SQLTYPECODE_BITVAR   = 15   /* not supported */
 };
 
+#define SQL_BOOLEAN			13
+
 /* #ifndef SQLDTCODE_DATE */
 /* specifies the type of datetime data type */
 enum SQLDATETIME_CODE {
```
Comment (selvaganesang) [inline win-odbc64/sql/cli/sqlcli.h]: Is this latest version of sqlci.h from sql/cli? If not consider copying the latest version of sqlcli.h to this folder or creating a softlink to this file.

Comment (xwq) [inline win-odbc64/sql/cli/sqlcli.h]: @selvaganesang Thanks for reminder. We shouldn't use 13 to define a new datatype here and we also don't need define a new datatype . We need use SQLTYPECODE_BOOLEAN as the extend datatype.

@Weixin-Xu Please change it.

**Pull request merged**